code/core/src/highlight/utils.ts

```diff
@@ -145,28 +141,68 @@ export const useStore = <T>(initialValue?: T) => {
   return { get, set, subscribe, teardown } as const;
 };
 
-export const mapElements = (highlights: HighlightOptions[]): Map<HTMLElement, Highlight> => {
+export const selectElements = (root: HTMLElement | null, selector: string | CallRef): Element[] => {
+  // CSS selector
+  if (typeof selector === 'string') {
+    return [
+      ...document.querySelectorAll(
+        // Elements matching the selector, excluding storybook elements and their descendants.
+        // Necessary to find portaled elements (e.g. children of `body`).
+        `:is(${selector}):not([id^="storybook-"], [id^="storybook-"] *, .sb-wrapper, .sb-wrapper *)`
```

Comment:
might make sense to turn this into a variable so it's easier to understand and update later if needed:
```
const STORYBOOK_ELEMENT_SELECTORS = `[id^="storybook-"], [id^="storybook-"] *, .sb-wrapper, .sb-wrapper *`
```